src/charts/ActiveAllocationByMonth.jsx

```diff
@@ -0,0 +1,292 @@
+									selectedYearFirst === 2023 || selectedYearFirst === '' ? 0 : data[7].map((i) => 1 - i.percentageAllocated).reduce((pv, cv) => pv + cv, 0).toFixed(2), 
+									data[8].map((i) => 1 - i.percentageAllocated).reduce((pv, cv) => pv + cv, 0).toFixed(2), 
+									data[9].map((i) => 1 - i.percentageAllocated).reduce((pv, cv) => pv + cv, 0).toFixed(2), 
+									data[10].map((i) => 1 - i.percentageAllocated).reduce((pv, cv) => pv + cv, 0).toFixed(2), 
+									data[11].map((i) => 1 - i.percentageAllocated).reduce((pv, cv) => pv + cv, 0).toFixed(2)
+								],
+								borderWidth: 1,
+								backgroundColor: ["#f08522"]
+							},
+					]
+				},
+				options: {
+					plugins: {
+						legend: {
+							position: 'bottom',
+						},
+						title: {
+							display: true,
+							text: 'Status de alocação mensal dos Projetos Contratados',
+							font: {
+								size: 18,
+							},
+							padding: {
+								top: 20,
+								bottom: 40
+							}
+						},
+						subtitle: {
+							display: true,
+							position: 'bottom',
+							text: '*Considerando alocação em qualquer período do mês',
+							font: {
+								size: 14,
+								family: 'tahoma',
+								weight: 'normal',
+								style: 'italic'
+							},
+							padding: {
+								bottom: 10
+							}
+						},
+						datalabels: {
+							anchor: 'end',
+							align: 'top',
+							display: function(context) {
+								return context.dataset.data[context.dataIndex] !== 0; // or >= 1 or ...
+						 	},
+							//formatter: Math.round,
+							font: {
+									weight: 'bold',
+									size: 14
+							}
+						}
+					},
+					scales: {
+						y: {
+								beginAtZero: true,
+								suggestedMax: areaMonthChart !== 'Todas' ? 20 : 50
+						},
+					},
+				}
+			});
+		}
+	};
+
+  const handleYearFirst = (event) => {
+		setSelectedYearFirst(event.target.value);
+		const data = handleMonthlyAllocation(event.target.value, selectedCentro);
+		setAreaMonthChart('');
+		drawBarChart(data, event.target.value)
+		console.log(event.target.value)
+	}
+
+  const filterMonthChart = async (event) => {
+		setAreaMonthChart(event.target.value);
+
+		let data = selectedCentro === 'CDM' ? monthlyAllocationCDM : monthlyAllocationICS;
+		if(event.target.value !== 'todas') {
+			data = data.map((month) => 
+				month.filter((i) => i["Área"] === event.target.value)
+			);
+		}
+		drawBarChart(data, selectedYearFirst);
+	}
+
+	const filterSubareaMonthChart = async (event) => {
+		setSubareaMonthChart(event.target.value);
+
+		let data = selectedCentro === 'CDM' ? monthlyAllocationCDM : monthlyAllocationICS;
+		if(event.target.value !== 'todas') {
+			data = data.map((month) => 
+				month.filter((i) => i["Função"] === event.target.value)
+			);
+		}
+		else {
+			data = data.map((month) => 
+				month.filter((i) => i["Área"] === areaMonthChart)
+			);
+		}
+		drawBarChart(data, selectedYearFirst);
+	}
+
+  useLayoutEffect(() => {
+    drawBarChart(monthlyAllocationCDM, selectedYearFirst)
```

Comment:
como esse componente só é renderizado quando o loading for false, já coloquei o drawBarChart no useLayoutEffect logo de cara. Aí ele renderiza o grafico passando como dado o CDM inicial.